src/data-requests/districts.ts

```diff
@@ -75,17 +80,30 @@ export async function getDistrictsRecoveredData(): Promise<
   });
   return {
     data: districts,
-    lastUpdate: new Date(data.features[0].attributes.date),
+    lastUpdate: parseDate(data.features[0].attributes.Datenstand),
   };
 }
 
 export async function getNewDistrictCases(): Promise<
   ResponseData<{ ags: string; cases: number }[]>
 > {
-  const response = await axios.get(
-    `https://services7.arcgis.com/mOBPykOjAyBO2ZKk/arcgis/rest/services/Covid19_hubv/FeatureServer/0/query?where=NeuerFall IN(1,-1)&objectIds=&time=&resultType=standard&outFields=AnzahlFall,MeldeDatum,IdLandkreis&returnIdsOnly=false&returnUniqueIdsOnly=false&returnCountOnly=false&returnDistinctValues=false&cacheHint=false&orderByFields=IdLandkreis&groupByFieldsForStatistics=IdLandkreis&outStatistics=[{"statisticType":"sum","onStatisticField":"AnzahlFall","outStatisticFieldName":"cases"},{"statisticType":"max","onStatisticField":"MeldeDatum","outStatisticFieldName":"date"}]&having=&resultOffset=&resultRecordCount=&sqlFormat=none&f=json&token=`
-  );
-  const data = response.data;
+  const url = `https://services7.arcgis.com/mOBPykOjAyBO2ZKk/arcgis/rest/services/Covid19_hubv/FeatureServer/0/query?where=NeuerFall IN(1,-1)&objectIds=&time=&resultType=standard&outFields=AnzahlFall,MeldeDatum,IdLandkreis,Datenstand&returnIdsOnly=false&returnUniqueIdsOnly=false&returnCountOnly=false&returnDistinctValues=false&cacheHint=false&orderByFields=IdLandkreis&groupByFieldsForStatistics=IdLandkreis,Datenstand&outStatistics=[{"statisticType":"sum","onStatisticField":"AnzahlFall","outStatisticFieldName":"cases"},{"statisticType":"max","onStatisticField":"MeldeDatum","outStatisticFieldName":"date"}]&having=&resultOffset=&resultRecordCount=&sqlFormat=none&f=json&token=`;
+  const response = await axios.get(url);
+  let data = response.data;
+  // check if data is updated, if not try alternate download from separately state tables
+  const now = new Date();
+  const nowTime = now.getTime(); //now im milliseconds
+  const actualDate = now.setHours(0, 0, 0, 0); // date 0:00 GMT
+  const threeOclock = now.setHours(3, 30, 0, 0); // after 3:30 GMT the RKI data update should be done
+  const Datenstand = parseDate(
+    data.features[0].attributes.Datenstand
+  ).getTime(); // Datenstand im milliseconds
+  if (
+    actualDate - Datenstand > 24 * 60 * 60000 ||
+    (Datenstand != actualDate && nowTime > threeOclock)
```

Comment:
Let's move this into a utility function because it is used quite often.
Something like `shouldUseAlternateDataSource(Datenstand: Date): boolean`

Reply:
done